apps/aecontract/src/aect_contracts_store.erl

```diff
@@ -41,14 +44,26 @@ new() ->
 
 -spec new(aeu_mtrees:mtree()) -> store().
 new(Tree) ->
-    #store{ cache = #{}, mtree = Tree }.
+    #store{ cache = #{}, read_cache = #{}, mtree = aeu_mp_trees:tree_no_cache(Tree) }.
 
 %% Returns empty binary if key is not in the store.
 -spec get(key(), store()) -> val().
-get(Key, #store{ cache = Cache, mtree = Tree }) ->
+get(Key, Store) ->
+    {Val, _NewStore} = get_w_cache(Key, Store),
+    Val.
+
+-spec get_w_cache(key(), store()) -> {val(), store()}.
+get_w_cache(Key, #store{ cache = Cache, read_cache = RCache, mtree = Tree } = S) ->
```

Comment:
If `get_w_cache/2` were moved to `aeu_mp_trees`, it could help us move towards producing block witnesses (where we should cache every read, and the read cache + write cache would make up the witness).

Reply:
That is an entirely different beast though :-)